smartsim/_core/generation/generator.py

```diff
@@ -199,33 +207,55 @@ def _build_commands(
 
         :param job: Job object
         :param job_path: The file path for the Job run folder
+        :param log_path: The file path for the Job log folder
         :return: A CommandList containing the file operation commands
         """
+        context = GenerationContext(job_path)
         cmd_list = CommandList()
-        cmd_list.commands.append(cls._mkdir_file(job_path))
-        cmd_list.commands.append(cls._mkdir_file(log_path))
-        entity = job.entity
+
+        cls._append_mkdir_commands(cmd_list, job_path, log_path)
+
         if isinstance(entity, _GenerableProtocol):
-            helpers: t.List[
-                t.Callable[
-                    [t.Union[EntityFiles, None], pathlib.Path],
-                    t.Union[CommandList, None],
-                ]
-            ] = [
-                cls._copy_files,
-                cls._symlink_files,
-                lambda files, path: cls._write_tagged_files(
-                    files, entity.file_parameters, path
-                ),
-            ]
-
-            for method in helpers:
-                return_cmd_list = method(entity.files, job_path)
-                if return_cmd_list:
-                    cmd_list.commands.extend(return_cmd_list.commands)
+            cls._append_file_operations(cmd_list, entity, context)
 
         return cmd_list
 
+    @classmethod
+    def _append_mkdir_commands(
+        cls, cmd_list: CommandList, job_path: pathlib.Path, log_path: pathlib.Path
+    ) -> None:
+        """Append file operation Commands (mkdir) for a Job's run and log directory.
+
+        :param cmd_list: A CommandList object containing the commands to be executed
+        :param job_path: The file path for the Job run folder
+        :param log_path: The file path for the Job log folder
+        """
+        cmd_list.commands.append(cls._mkdir_file(job_path))
+        cmd_list.commands.append(cls._mkdir_file(log_path))
+
+    @classmethod
+    def _append_file_operations(
+        cls,
+        cmd_list: CommandList,
+        entity: _GenerableProtocol,
+        context: GenerationContext,
+    ) -> None:
+        """Append file operation Commands (copy, symlink, configure) for all
+        files attached to the entity.
+
+        :param cmd_list: A CommandList object containing the commands to be executed
+        :param entity: The Job's attached entity
+        :param context: A GenerationContext object that holds the Job's run directory
+        """
+        copy_ret = cls._copy_files(entity.files.copy_operations, context)
+        cmd_list.commands.extend(copy_ret.commands)
+
+        symlink_ret = cls._symlink_files(entity.files.symlink_operations, context)
+        cmd_list.commands.extend(symlink_ret.commands)
+
+        configure_ret = cls._configure_files(entity.files.configure_operations, context)
+        cmd_list.commands.extend(configure_ret.commands)
```

Comment:
Similar "use command list directly on off chance we change the internal structure" question here:
```suggestion
        copy_ret = cls._copy_files(entity.files.copy_operations, context)
        cmd_list.extend(copy_ret)

        symlink_ret = cls._symlink_files(entity.files.symlink_operations, context)
        cmd_list.extend(symlink_ret)

        configure_ret = cls._configure_files(entity.files.configure_operations, context)
        cmd_list.extend(configure_ret)
```